stanford_news.module

```diff
@@ -0,0 +1,168 @@
+    $target = $token_service->replace("[node:su_news_source:uri]", ['node' => $entity]);
+
+    // Check if the token that was provided is actually a url. If it is, then we
+    // can set the rabbit hole action to redirect to that url.
+    if (is_string($target) && UrlHelper::isValid($target, TRUE)) {
+      $entity->set('rh_action', 'page_redirect');
+      $entity->set('rh_redirect', $target);
+      $entity->set('rh_redirect_response', 301);
+    }
+    else {
+      $entity->set('rh_action', 'bundle_default');
+      $entity->set('rh_redirect', '');
+    }
+  }
+
+  // If a news item is being edited or saved, clear out some cache tags.
+  if ($entity->bundle() == "stanford_news") {
+    Cache::invalidateTags(["node_list:{$entity->bundle()}"]);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function stanford_news_node_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+
+  // Only run on stanford news items.
+  if ($entity->bundle() !== "stanford_news" || !node_is_page($entity)) {
+    return;
+  }
+
+  $rh_action = $entity->get('rh_action')->getValue()[0]['value'] ?? FALSE;
+  $rh_redirect = $entity->get('rh_redirect')->getValue()[0]['value'] ?? FALSE;
+
+  // Display a message to the user that this page would redirect to another
+  // location for other users.
+  if ($rh_action == "page_redirect" && !empty($rh_redirect)) {
+    \Drupal::messenger()
+      ->addWarning(t('This page will redirect to @url for other visitors.', ['@url' => $rh_redirect]));
+  }
+
+}
+
+/**
+ * Implements hook_views_post_render().
+ *
+ * Views render arrays contain a cache tag "node_list". This cache tag is
+ * cleared every time ANY node is created, edited or deleted. When this happens
+ * every view on the site gets its cache flushed. This causes poor performance
+ * since a view would get flushed even if it has no relation to that node. To
+ * assist in cache tags, we create a custom cache tag based on the node type
+ * filter on the view. Its a small improvement but will have huge impact in
+ * keeping cached renders much longer.
+ *
+ * @see stanford_person_node_presave()
+ * @see stanford_person_taxonomy_term_presave ()
+ */
+function stanford_news_views_post_render(ViewExecutable $view, &$output, CachePluginBase $cache) {
+
+  // Node Base Table Views.
+  if ($view->storage->id() == 'stanford_news') {
+    $output['#attached']['library'][] = 'stanford_news/news_list';
+
+    $node_list_position = array_search('node_list', $output['#cache']['tags']);
+    unset($output['#cache']['tags'][$node_list_position]);
+    foreach ($view->filter['type']->value as $node_type) {
+      $output['#cache']['tags'][] = "node_list:$node_type";
+    }
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function stanford_news_form_node_stanford_news_form_alter(&$form, FormStateInterface $form_state) {
+  $form['su_news_topics']['widget']['add_more']['#value'] = t("Add another term");
```

Comment:
This feels like it would be better as a `hook_field_widget_form_alter()`

Reply:
I don't really want to change ALL term widgets. Just the one on this particular node. This module would be overreaching if I changed all form widgets. I suppose I could get the current form from the `$context` parameter in hook_field_widget_form_alter() but that seems like it is more overhead than this alter. Thoughts?

Reply:
yeah the `$context` would have the field you would target. Its probably the same amount of overhead. I think its better to use a hook with as narrowed focus as possible.

Reply:
To quote John, _FINE_

Reply:
I think that's a Pookie quote.